src/payment/checkout/payment-form/PaymentForm.jsx

```diff
@@ -222,7 +199,7 @@ export class PaymentFormComponent extends React.Component {
                   type="submit"
                   className="btn btn-primary btn-lg btn-block"
                   state={submitButtonState}
-                  onClick={this.handleSubmitButtonClick}
+                  onClick={this.props.onSubmitButtonClick}
```

Comment:
Hm!  Continuing the thought from above... So the `onSubmitButtonClick` and `onSubmitPayment` handlers do basically fire at the same time... I wonder if we could effectively remove the one on this button and just rely on the form `onSubmit`?  On the outside, the handler would just do both things.  Might be easier to understand for others.